### PR TITLE
Convert more unary functions

### PR DESCRIPTION
**Pull request by petrosagg**
### Testing

- [ ] This PR has adequate test coverage / QA involvement has been duly considered.

### Release notes

This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):

  -

### CODE REVIEW

**Review by aalexandrov — APPROVED** [review]
Looks good.

File: src/expr/src/scalar/func/impls/byte.rs [hunk 1]
```diff
@@ -18,3 +20,17 @@ sqlfunc!(
...
+    #[sqlname = "bit_length"]
+    fn bit_length_bytes<'a>(a: &'a [u8]) -> Result<i32, EvalError> {
+        i32::try_from(a.len() * 8).or(Err(EvalError::Int32OutOfRange))
+    }
+);
+
+sqlfunc!(
+    #[sqlname = "octet_length"]
+    fn byte_length_bytes<'a>(a: &'a [u8]) -> Result<i32, EvalError> {
+        i32::try_from(a.len()).or(Err(EvalError::Int32OutOfRange))
```
[the remark this answers is not in the document]
Comment (petrosagg) [reply src/expr/src/scalar/func/impls/byte.rs:34]: Yes, that's exactly the intuition

File: src/expr/src/scalar/func/impls/jsonb.rs [hunk 2]
```diff
@@ -178,3 +166,102 @@ sqlfunc!(
...
+            Datum::JsonNull => "null",
+            d => panic!("Not jsonb: {:?}", d),
+        }
+    }
+);
+
+#[derive(Ord, PartialOrd, Clone, Debug, Eq, PartialEq, Serialize, Deserialize, Hash, MzReflect)]
+pub struct JsonbStripNulls;
+
+impl LazyUnaryFunc for JsonbStripNulls {
```
[the remark this answers is not in the document]
Comment (petrosagg) [reply src/expr/src/scalar/func/impls/jsonb.rs]: It just allows it to re-use the `temporary_storage` which is more efficient but I switched it now to `sqlfunc!` which is  a lot simpler

File: src/expr/src/scalar/func/impls/byte.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (petrosagg) [reply src/expr/src/scalar/func/impls/byte.rs:34]: Also see this relevant clippy discussion on whether any const function should be allowed in an `.or(..)` (the answer no) https://github.com/rust-lang/rust-clippy/issues/5658

**Review by sploiselle — APPROVED** [review]
LGTM

**Pull request merged**